<big><b>Step 1:</b> Load the SQL extension</big>

In [ ]:
%load_ext sql

<big><b>Step 2</b>: Connect sqlite to coursedatawarehouse database</big>

In [24]:
%%sql

sqlite:///CourseDataWarehouseDemo.db

'Connected: @CourseDataWarehouseDemo.db'

<big><b>Step 3</b>: Query test</big>

<b> Scenario 1 <b>: Finding the details of which instructor taught the course that meets the most times in Spring 2019. 

In [73]:
%%sql

SELECT I.IID, I.Name, T.Term, CC.Title, COUNT(CC.Title)/2 AS Cnt
FROM REG_COURSE_WH AS RC
    JOIN INSTRUCTOR_WH AS I ON (RC.IID = I.IID)
    JOIN TERM_WH AS T ON (RC.TermID = T.TermID)
    JOIN CATALOG_COURSE_WH AS CC ON (CC.CAID = RC.CAID)
WHERE Term = 'Spring2019'
GROUP BY I.Name, Term
ORDER BY Cnt Desc
LIMIT 5;

   sqlite:///CourseData.db
 * sqlite:///CourseDataWarehouseDemo.db
Done.


[(892, 'Ellen M.  Lee', 'Spring2019', 'Romantic Love in Greek and Roman Literature', 152),
 (70, 'Sharlene A.  McEvoy', 'Spring2019', 'Legal Environment of Business', 150),
 (552, 'Anne E  Campbell', 'Spring2019', 'World Language Methods', 130),
 (274, 'Vishnu V.  Vinekar', 'Spring2019', 'Introduction to Information Systems', 130),
 (577, 'Jillian E.  Smith-Carpenter', 'Spring2019', 'Biochemistry I', 127)]

<b>What other courses did this instructor teach in Spring 2019?</b>

In [74]:
%%sql

SELECT DISTINCT CC.Title, I.Name, Term
FROM REG_COURSE_WH AS RC
    JOIN INSTRUCTOR_WH AS I ON (RC.IID = I.IID)
    JOIN CATALOG_COURSE_WH AS CC ON (CC.CAID = RC.CAID)
    JOIN TERM_WH AS T ON (RC.TermID = T.TermID)
WHERE I.IID = 892 AND T.Term = 'Spring2019';

   sqlite:///CourseData.db
 * sqlite:///CourseDataWarehouseDemo.db
Done.


[('Romantic Love in Greek and Roman Literature', 'Ellen M.  Lee', 'Spring2019'),
 ('Honors Seminar', 'Ellen M.  Lee', 'Spring2019'),
 ('Readings in Latin Prose and Poetry II', 'Ellen M.  Lee', 'Spring2019')]

<b>What was the average remaining number of open spots in Readings in Latin Prose and Poetry II? </b>

In [83]:
%%sql

SELECT CC.Title, AVG(RC.Cap) AS Max, AVG(RC.Act) AS Filled, AVG(RC.Rem) AS Open
FROM REG_COURSE_WH AS RC
       JOIN CATALOG_COURSE_WH AS CC ON (CC.CAID = RC.CAID)
WHERE CC.Title = 'Readings in Latin Prose and Poetry II'
GROUP BY CC.Title;

   sqlite:///CourseData.db
 * sqlite:///CourseDataWarehouseDemo.db
Done.


[('Readings in Latin Prose and Poetry II', 23.383084577114428, 19.855721393034827, 3.527363184079602)]

<b>We can conclude that there is an average of ~23 spots in each Readings in Latin Prose and Poetry II class, with an average of 20 spots taken, and an average of 3.5 spots open.